utils.js

```diff
@@ -923,10 +1096,13 @@ TrezorSigner.prototype.restore = function (password) {
   if (this.Signer.accountIndex > 1000) {
     return false
   }
-  this.Signer.accounts = decryptedData.accounts
+
   this.Signer.changeIndex = -1
   this.Signer.receivingIndex = -1
-  this.Signer.accountIndex = 0
+  this.Signer.accountIndex = decryptedData.numXpubs
+  for (let i = 0; i <= this.Signer.accountIndex; i++) {
+    this.Signer.accounts.push(new BIP84.fromZPub(decryptedData.xpubArr[i], this.Signer.pubTypes, this.Signer.networks))
```

Comment:
should we serialize pubtypes and networks so there isn't a case where you serialize with one and deserialize and get corrupted accounts on another set of pubtypes/networks?